src/validation.cpp

```diff
@@ -3629,25 +3621,32 @@ static bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &pos,
 
 static bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, const Consensus::Params& consensusParams, bool fCheckPOW = true)
 {
-    /*
-     *
-     *   Header verification takes a long time due to having to compute scrypt hashes.
-     * This is a lengthy process, and so for this reason we skip it. This is acceptable
-     * since we're going to run a full validation of each and every block in the chain
-     * once the full data set is received.
-     *
-     */
-
+    // Check header only on recieving blocks
     return true;
 }
 
-static bool CheckBlockHeaderWorkHash(const CBlockHeader& block, CValidationState& state, const Consensus::Params& consensusParams, bool fCheckPOW = true)
+static bool CheckWorkBlockHeader(const CBlockHeader& block, CValidationState& state, const Consensus::Params& consensusParams, bool fCheckPOW = true)
 {
-    // Check proof of work matches claimed amount
-    if (!IsBlockchainX16R() && fCheckPOW && 
-        (!CheckProofOfWork(block.GetNextMinedHash(), block.nBits, consensusParams) ||
-         !CheckProofOfWork(block.GetWorkHash(), block.nBits, consensusParams) )) 
-        return state.DoS(50, false, REJECT_INVALID, "high-hash", false, block.GetNextMinedHash().ToString() + " proof of work failed " + block.GetWorkHash().ToString());
+    CBlockIndex* pindexPrev = chainActive.Tip();
```

Comment:
I propose to use height of block.previousBlockHash instead of chainActive.Tip().height

Reply:
\+